src/storage/memory/transaction.rs

```diff
@@ -104,26 +102,13 @@ impl Transaction for InMemoryTransaction {
     // TODO: remove this unused variable
     async fn scan(
         &self,
-        begin_sort_key: &[DataValue],
-        end_sort_key: &[DataValue],
         col_idx: &[StorageColumnRef],
-        is_sorted: bool,
-        reversed: bool,
-        expr: Option<BoundExpr>,
+        opts: ScanOptions,
     ) -> StorageResult<InMemoryTxnIterator> {
-        assert!(expr.is_none(), "MemTxn doesn't support filter scan");
-        assert!(!reversed, "reverse iterator is not supported for now");
-
-        assert!(
-            begin_sort_key.is_empty(),
-            "sort_key is not supported in InMemoryEngine for now"
-        );
-        assert!(
-            end_sort_key.is_empty(),
-            "sort_key is not supported in InMemoryEngine for now"
-        );
-
-        let snapshot = if is_sorted {
+        assert!(opts.filter.is_none(), "MemTxn doesn't support filter scan");
```

Comment:
Do we plan to support range-filter scan (store in order by key) in memory storage?